feat: add `file:` protocol path support

https://docs.rs/url/latest/url/struct.Url.html#method.to_file_path

`file:` protocol is required to be used on Windows for `import` statement.

It seems impossible to implement it due to WASM limitation:

```log
error[E0599]: no method named `to_file_path` found for struct `Url` in the current scope
   --> src/lib.rs:394:18
    |
392 |               let path = Url::parse(specifier)
    |  ________________________-
3[93](https://github.com/unrs/unrs-resolver/actions/runs/15589914554/job/43906035982?pr=151#step:4:94) | |                 .map_err(|_| unsupported_error.clone())?
3[94](https://github.com/unrs/unrs-resolver/actions/runs/15589914554/job/43906035982?pr=151#step:4:95) | |                 .to_file_path()
    | |                 -^^^^^^^^^^^^ method not found in `Url`
    | |_________________|
    |
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -77,8 +77,8 @@ jobs:
 
       - name: Check
         run: |
-          rustup target add wasm32-unknown-unknown
-          cargo check --all-features --target wasm32-unknown-unknown
+          rustup target add wasm32-wasip1
+          cargo check --all-features --target wasm32-wasip1
 
   wasi:
     name: Test wasi target

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -92,6 +92,7 @@ serde_json = { version = "1.0.140", features = ["preserve_order"], optional = tr
 simdutf8 = { version = "0.1.5", features = ["aarch64_neon"] }
 thiserror = "2.0.12"
 tracing = "0.1.41"
+url = "2.5.4"
 windows-sys = { version = "0.59", features = ["Win32_Foundation", "Win32_Storage_FileSystem"] }
 
 pnp = { version = "0.9.4", optional = true }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -81,6 +81,7 @@ use std::{
     path::{Component, Path, PathBuf},
     sync::Arc,
 };
+use url::Url;
 
 #[cfg(feature = "fs_cache")]
 pub use crate::{
@@ -379,6 +380,23 @@ impl<C: Cache<Cp = FsCachedPath>> ResolverGeneric<C> {
         // enhanced-resolve: try alias
         if let Some(path) = self.load_alias(cached_path, specifier, &self.options.alias, ctx)? {
             return Ok(path);
+        }
+
+        #[allow(unused_assignments)]
+        let mut specifier_owned: Option<String> = None;
+        let mut specifier = specifier;
+
+        if specifier.starts_with("file://") {
+            let unsupported_error = ResolveError::PathNotSupported(specifier.into());
+
+            let path = Url::parse(specifier)
+                .map_err(|_| unsupported_error.clone())?
+                .to_file_path()
+                .map_err(|()| unsupported_error)?;
+
+            let owned = path.to_string_lossy().into_owned();
+            specifier_owned = Some(owned);
+            specifier = specifier_owned.as_deref().unwrap();
         }
 
         let result = match Path::new(specifier).components().next() {

--- a/src/tests/resolve.rs
+++ b/src/tests/resolve.rs
@@ -1,5 +1,7 @@
 //! <https://github.com/webpack/enhanced-resolve/blob/main/test/resolve.test.js>
 
+use url::Url;
+
 use crate::{Resolution, ResolveError, ResolveOptions, Resolver};
 
 #[test]
@@ -10,9 +12,12 @@ fn resolve() {
 
     let main1_js_path = f.join("main1.js").to_string_lossy().to_string();
 
+    let file_protocol_path = Url::from_file_path(main1_js_path.clone()).unwrap();
+
     #[rustfmt::skip]
     let pass = [
         ("absolute path", f.clone(), main1_js_path.as_str(), f.join("main1.js")),
+        ("file protocol absolute path", f.clone(), file_protocol_path.as_str(), f.join("main1.js")),
         ("file with .js", f.clone(), "./main1.js", f.join("main1.js")),
         ("file without extension", f.clone(), "./main1", f.join("main1.js")),
         ("another file with .js", f.clone(), "./a.js", f.join("a.js")),
@@ -62,6 +67,13 @@ fn resolve() {
         }
         assert_eq!(resolved_path, Some(expected), "{comment} {path:?} {request}");
     }
+
+    #[cfg(windows)]
+    let resolve_error = ResolveError::NotFound("\\\\.\\main.js".into());
+    #[cfg(not(windows))]
+    let resolve_error = ResolveError::PathNotSupported("file://./main.js".into());
+
+    assert_eq!(resolver.resolve(f, "file://./main.js"), Err(resolve_error));
 }
 
 #[test]

